assets/queries/openAPI/2.0/unknown_property/query.rego

```diff
@@ -0,0 +1,301 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) == "2.0"
+
+	[path, value] := walk(doc)
+
+	field := path[0]
+	all([field != "id", field != "file"])
+	not known_swagger_object_field(field)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s", [field]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("The field '%s' is known in the openapi object", [field]),
+		"keyActualValue": sprintf("The field '%s' is unknown in the openapi object", [field]),
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) == "2.0"
+
+	[path, value] := walk(doc)
+
+	object := simple_objects[o]
+	path[minus(count(path), 1)] == o
+
+	v := value[x]
+	not known_field(object, x)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.%s", [openapi_lib.concat_path(path), x]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("The field '%s' is known in the %s object", [x, o]),
+		"keyActualValue": sprintf("The field '%s' is unknown in the %s object", [x, o]),
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) == "2.0"
+
+	[path, value] := walk(doc)
+
+	object := array_objects[o]
+	path[minus(count(path), 1)] == o
```

Comment:
Same comment as above, also I think, maybe, it is interesting to extract a method of this two lines...